mu/logic.py

```diff
@@ -234,6 +234,55 @@ def check_pycodestyle(code):
     return style_feedback
 
 
+def get_runtime_hex_path():
+    """
+    Returns the path to the hex runtime file - if this has been
+    specified under element 'runtime_hex' in settings.json.
+    This can be a fully-qualified file path, or just a file name
+    in which case the file should be located in the workspace directory.
+    """
+    runtime_hex_path = ""
+    settings_path = get_settings_path()
+    settings = {}
+    try:
+        with open(settings_path) as f:
+            settings = json.load(f)
+    except FileNotFoundError:
+        logger.error('Settings file {} does not exist.'.format(settings_path))
+    except ValueError:
+        logger.error('Settings file {} could not be parsed.'.format(
+                     settings_path))
+    else:
+        if 'runtime_hex' in settings and settings['runtime_hex'] != "":
+            runtime_hex_path = os.path.join(
+                get_workspace_dir(),
+                settings['runtime_hex'])
+
+    return runtime_hex_path
+
+
+def get_runtime_hex_str():
```

Comment:
Is this this function really necessary? uFlash itself can take the path to a runtime hex file, so it'd be better not to duplicate efforts and pass that to uFlash directly